### PR TITLE
macOS lazily saves the AVX512 context

### LINKED ISSUE #36329: LLVM-9.0.1 doesn't recognize AVX-512 in macOS
See https://github.com/SciML/DifferentialEquations.jl/issues/623#issuecomment-644739630

@maleadt mentioned that https://github.com/llvm/llvm-project/commit/82921bf2baed96b700f90b090d5dc2530223d9c0 might fix it. I wonder if we can patch Julia's LLVM to make it generate more efficient code for new Apple hardware.

### PR DESCRIPTION
**Pull request by YingboMa**
Fix #36329

### CODE REVIEW

File: src/processor_x86.cpp [hunk 1]
```diff
@@ -545,7 +545,13 @@ static NOINLINE std::pair<uint32_t,FeatureList<feature_sz>> _get_host_cpu(void)
     unset_bits(features, 32 + 27);
     if (!hasavx)
         features_disable_avx(features);
+#ifdef OS_DARWIN
```
Comment (yuyichao) [inline src/processor_x86.cpp:548]: `_OS_DARWIN_` i believe.

File: src/processor_x86.cpp [hunk 2]
```diff
@@ -545,7 +545,13 @@ static NOINLINE std::pair<uint32_t,FeatureList<feature_sz>> _get_host_cpu(void)
...
+    // See https://github.com/llvm/llvm-project/commit/82921bf2baed96b700f90b090d5dc2530223d9c0
+    // Darwin lazily saves the AVX512 context on first use
```
Comment (yuyichao) [inline src/processor_x86.cpp:550]: also https://github.com/apple/darwin-xnu/blob/a449c6a3b8014d9406c2ddbdc81795da24aa7443/osfmk/i386/fpu.c#L174 or whatever appropriate direct document from apple.

**Review by yuyichao — CHANGES_REQUESTED** [review]

**Review by yuyichao — APPROVED** [review]
LGTM. Do note that since it seems that loop vectorizer might be involved there'll still be other issues that you'll need to hack around until it is moved to the LLVM level.

**Comment by vchuravy** [comment]
We should also back-port the llvm patch.

**Comment by Keno** [comment]
This seems like a terrible idea on their part, but I guess it's what we have to do.

**Comment by yuyichao** [comment]
IIUC it sounds like they are reimplementing `xsave`. It's pretty unfortunate that they deviate from the intel document...

I assume it also means that running a darwin vm that doesn't support avx512 on hardware that does is basically unsupported then... (which is probably something that apple couldn't care less about...)

**Comment by Keno** [comment]
I doubt this is the last we've heard along these lines, but let's try this for now, since it matches LLVM.

**Comment by YingboMa** [comment]
Is it possible to back port this change?

**Pull request merged**